projects/Button_ADC/src/main.c

```diff
@@ -0,0 +1,53 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+static GpioAddress button_address = { .port = GPIO_PORT_B, .pin = 2 };
+
+static GpioAddress adc_address = { .port = GPIO_PORT_A, .pin = 6 };
+
+static GpioSettings adc_settings = {
+  .direction = GPIO_DIR_IN,
+  .resistor = GPIO_RES_NONE,
+  .alt_function = GPIO_ALTFN_ANALOG,
+};
+
+static InterruptSettings interrupt_settings = {
+  .type = INTERRUPT_TYPE_INTERRUPT,
+  .priority = INTERRUPT_PRIORITY_NORMAL,
+};
+
+static AdcChannel adc_pinA6_channel = ADC_CHANNEL_6;
+
+static uint16_t channel_data = 0;
```

Comment:
Please prefix static variables with `s_`, that way we can tell that they're static more easily. That being said, these global variables could be local variables, especially `channel_data`. You can avoid having a global variable for `adc_pinA6_channel` by passing a pointer to it around through the `context` parameter on interrupt callbacks; currently you're passing a pointer to `adc_address` around but not using it. It's good practice to minimize the use of global variables as much as possible.